From 85b64b03d7997240c7e8974980ecefcc9ab9cd7c Mon Sep 17 00:00:00 2001
From: Michal Suchanek <msuchanek@suse.de>
Date: Tue, 11 Apr 2017 23:09:48 +0200
Subject: [PATCH] ibmvnic: use max_mtu instead of req_mtu for MTU range check.

References: bsc#1031512
Patch-mainline: never - not needed
Upstream commit d894be57ca92c8a8819ab544d550809e8731137b
("ethernet: use net core MTU range checking in more drivers") besides
using upstream feature we don't have corrects mtu range checking.

Signed-off-by: Michal Suchanek <msuchanek@suse.de>
---
 drivers/net/ethernet/ibm/ibmvnic.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/ibm/ibmvnic.c b/drivers/net/ethernet/ibm/ibmvnic.c
index e807317..07ff6e6 100644
--- a/drivers/net/ethernet/ibm/ibmvnic.c
+++ b/drivers/net/ethernet/ibm/ibmvnic.c
@@ -905,7 +905,7 @@ static int ibmvnic_change_mtu(struct net_device *netdev, int new_mtu)
 {
 	struct ibmvnic_adapter *adapter = netdev_priv(netdev);
 
-	if (new_mtu > adapter->req_mtu || new_mtu < adapter->min_mtu)
+	if (new_mtu > adapter->max_mtu || new_mtu < adapter->min_mtu)
 		return -EINVAL;
 
 	netdev->mtu = new_mtu;
-- 
2.10.2

